tests/unit/dataactvalidator/test_c9_award_financial.py

```diff
@@ -120,11 +120,21 @@ def test_zero_federal_action_obligation_and_positive_original_loan_subsidy_cost(
 
     tas = _TAS
     afa = AwardFinancialAssistanceFactory(tas=tas, fain='abc', uri=None, federal_action_obligation=0,
-                                          original_loan_subsidy_cost='1')
+                                          original_loan_subsidy_cost='1', assistance_type='08')
+    afa_2 = AwardFinancialAssistanceFactory(tas=tas, fain='abc', uri=None, federal_action_obligation=2,
+                                            original_loan_subsidy_cost='1', assistance_type='07')
+    af = AwardFinancialFactory(tas=tas, submisson_id=afa.submission_id, fain=None, uri=None)
+
+    errors = number_of_errors(_FILE, database, models=[afa, af, afa_2])
+    assert errors == 2
+
+    # Test that this is ignored if assistance type is 07
+    afa = AwardFinancialAssistanceFactory(tas=tas, fain='abc', uri=None, federal_action_obligation=0,
+                                          original_loan_subsidy_cost='1', assistance_type='07')
     af = AwardFinancialFactory(tas=tas, submisson_id=afa.submission_id, fain=None, uri=None)
 
     errors = number_of_errors(_FILE, database, models=[afa, af])
-    assert errors == 1
+    assert errors == 0
 
 
 def test_positive_federal_action_obligation_and_zero_original_loan_subsidy_cost(database):
```

Comment:
This function's name makes it seem that you are only testing a positive federal action obligation and a 0 loan subsidy cost, but you are testing loan costs of negative and positive values.